bee-network/bee-autopeering/src/peering/manager.rs

```diff
@@ -0,0 +1,774 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    filter::NeighborFilter,
+    messages::{DropPeeringRequest, PeeringRequest, PeeringResponse},
+    neighbor::{self, Neighborhood, SIZE_INBOUND, SIZE_OUTBOUND},
+};
+
+use crate::{
+    event::{Event, EventTx},
+    local::{
+        salt::{self, Salt, SALT_LIFETIME_SECS},
+        services::AUTOPEERING_SERVICE_NAME,
+        Local,
+    },
+    packet::{msg_hash, IncomingPacket, MessageType, OutgoingPacket},
+    peer::{self, lists::ActivePeersList, peer_id::PeerId, Peer},
+    peering::neighbor::{salt_distance, Neighbor},
+    request::{self, RequestManager, RequestValue, ResponseTx, RESPONSE_TIMEOUT},
+    server::{ServerSocket, ServerTx},
+    task::{Repeat, Runnable, ShutdownRx},
+    time::SECOND,
+    NeighborValidator,
+};
+
+use std::{net::SocketAddr, time::Duration};
+
+/// Salt update interval.
+pub(crate) const SALT_UPDATE_SECS: Duration = Duration::from_secs(SALT_LIFETIME_SECS.as_secs() - SECOND);
+const INCOMING: bool = true;
+const OUTGOING: bool = false;
+
+pub(crate) type InboundNeighborhood = Neighborhood<SIZE_INBOUND, true>;
+pub(crate) type OutboundNeighborhood = Neighborhood<SIZE_OUTBOUND, false>;
```

Comment:
Should these use INCOMING and OUTGOING ?

Reply:
yup. Done.